database/migrations/2020_10_29_160742_create_books_table.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\Schema;
+
+class CreateBooksTable extends Migration
+{
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('books', function (Blueprint $table) {
+            $table->id();
+            $table->string('name');
+            $table->string('slug');
+            $table->string('image');
+            $table->unsignedBigInteger('publication_id');
+            $table->string('isbn_number');
+            $table->integer('total_copies');
+            $table->integer('available_copies');
+            $table->foreign('publication_id')->references('id')->on('publications');
+            $table->softDeletes('deleted_at', 0);
+            $table->timestamps();
+        });
+    }
+
+    /**
+     * Reverse the migrations.
+     *
+     * @return void
+     */
+    public function down()
+    {
+        Schema::dropIfExists('books');
```

Comment:
You need to drop the 'foreign key relationship' first, then drop the table.

Reply:
Fix it for other tables too.

Reply:
done